src/main/java/com/petros/bring/bean/factory/AnnotationBeanFactory.java

```diff
@@ -27,6 +27,10 @@ public AnnotationBeanFactory(BeanDefinitionRegistry registry, List<BeanPostProce
 
     @Override
     public <T> T getBean(Class<T> beanType) throws NoSuchBeanException, NoUniqueBeanException {
+        Set<BeanDefinition> beanDefinitions = registry.getBeanDefinitionByType(beanType);
+        if (beanDefinitions.size() > 1 ) {
+            ensureBeanDefinitionsCreated(beanDefinitions);
```

Comment:
Looks like, here we create beans and method createBean(Class<?> clazz) is never used.